2024.03/alpine/Dockerfile

```diff
@@ -1,9 +1,10 @@
-FROM alpine:3.19
+FROM alpine:latest
```

Comment:
I would prefer we keep explicit versions, at least for now.

There are a variety of well-reasoned arguments out there against using `latest` tags for production. A simple case to consider would be if Alpine were to update after we publish an image, someone who builds the docker image themselves is going to have a different version of the OS to someone who pulls the published one.

I believe it would make building these images simpler, but I think we should open an issue instead and have a discussion about potential unforeseen consequences of a change like this.

It may be a better approach for us to adapt conventions used by other Docker Official images, I believe there are some which have scripts set up for some automation:

https://github.com/docker-library/ruby
https://github.com/docker-library/golang

Reply:
> I would prefer we keep explicit versions, at least for now.

Sorry, but I have to disagree here. Explicit versions are inherently technical debt. Sooner or later, you'll have to pay back. `latest` is safe by definition, and if there happens to be a vulnerability with the version you've built, just build it again. 
> 
> There are a variety of well-reasoned arguments out there against using `latest` tags for production. A simple case to consider would be if Alpine were to update after we publish an image, someone who builds the docker image themselves is going to have a different version of the OS to someone who pulls the published one.

The workflow can be built to be triggered on demand, or simply schedule updating after certain time. Plus the OS version here is not such a big issue, since no API is being used directly. As a matter of fact, scheduling a build under latest would be advisable, just in case there's some regression.

> I believe it would make building these images simpler, but I think we should open an issue instead and have a discussion about potential unforeseen consequences of a change like this.

Opening an issue before a PR is always a reasonable option. I'm happy to move the discussion there, if you so wish.

> It may be a better approach for us to adapt conventions used by other Docker Official images, I believe there are some which have scripts set up for some automation:
> 
> https://github.com/docker-library/ruby https://github.com/docker-library/golang

Well, I'm not so fond of official images myself. They're one size fits all. But as said above, automation will always save you time.

Reply:
I'm a huge fan of reproducible builds... for "enterprise software". That's why I'm pushing Rakudo dev's to "git tag" their modules with the same version info as they have in the `META6.json` files... so that's a point for "explicit versions".

In this case here, we talk about docker containers. Per definition, those should have a very short run (and life ;) ) time. And also, from my understanding, "latest" is stable and something with a good reason. There wouldn't be a need for "latest" if the previous "explicit" version wouldn't have issues or is somehow outdated.
With explicit versions, we would need to also define when such a version is good to be used for our docker build.... something like:
- it's already there for XYZ months and/or
- it's downloaded / installed /used XYZ thousand times
Also, don't know if this really happens with containers, but it does with "enterprise software releases", what if an implicit version, which we just have decided to use, will be discontinued for a good reason (say "security" or "bug" un general). How will we act here then?

So, for our case here, say short living containers, I would feel "ok" with "latest"...

Reply:
... and whatever we define, it has to be reflected in our automation, CI/CD, workflows... because doing the docker releases manually is nothing I'll vote / go for!